myia/compile/backends/relay.py

```diff
@@ -263,7 +263,7 @@ def relay_argmax(c, v, dims):
     """Implementation of argmax for Relay."""
     v = c.ref(v)
     assert dims.is_constant(tuple)
-    return relay.cast(relay.argmax(v, axis=dims.value), "int64")
+    return relay.cast(relay.argmax(v, axis=dims.value, keepdims=True), "int64")
```

Comment:
I'm not sure an unconditional keepdims is ok here.

I'm also not sure what behaviour we expect out of argmax wrt reduced dimension for our op.

Reply:
Hi @abergeron ! It seems `infer_shape` function for primitive argmax expects all dimensions to be here and reduced dimensions with size 1: https://github.com/mila-iqia/myia/blob/master/myia/operations/prim_argmax.py#L17

So, currently, I guess keepdims is necessary.

Reply:
ok